cloudify_deployment_proxy/polling.py

```diff
@@ -190,14 +191,29 @@ def dep_workflow_in_state_pollster(_client,
             'Executions list failed {0}.'.format(str(ex)))
     else:
         for _exec in _execs:
+            ctx.logger.debug(
+                '_exec info id {0}'.format(_exec.get('id')))
+
+            ctx.logger.debug(
+                '_exec info state {0}'.format(_exec.get('status')))
+
             if _workflow_id and _exec.get('workflow_id', '') != _workflow_id:
                 continue
             if _log_redirect:
                 ctx.logger.debug(
                     '_exec info for _log_redirect is {0}'.format(_exec))
                 dep_logs_redirect(_client, _exec.get('id'))
+
             if _exec.get('status') == _state:
-                return True
+                # After checking the status, another check needed which is
+                # to check if the ``_execution_id`` is set or not and
+                # whether or not the ``_execution_id`` is the same as the
+                # one returned from the api
+                if not _execution_id or _execution_id == _exec.get('id'):
```

Comment:
Now the best solution will be to make attempt of taking single execution with given execution_id instead of iteration through execution list. So you can replace this call in 183 line:

_client.executions.list(deployment_id=_dep_id,
                                     _include=exec_list_fields)
With something like this:

execution = _client.executions.get(
    deployment_id=_dep_id,
    execution_id=_execution_id
     _include=exec_list_fields)

And then (instead of loop): 
 if no execution --> raise NonRecoverableError or RuntimeError
 else: check status (workflow ID should be ok)

Reply:
I think you are right but in in case with create_deployment action we cannot get execution_id  so that we need to keep using the list instead of get

https://github.com/cloudify-incubator/cloudify-utilities-plugin/pull/88/files#diff-f866ca02694f54222a0c0869a0291159L176

his api does not return execution_id

so we cannot use execution_id here

Reply:
I did update it and it works now